echopop/computation/biology.py

```diff
@@ -104,160 +128,751 @@ def sum_strata_weight( haul_data: pd.DataFrame ,
+    )   
+
+    # ---- Remove unnecessary columns and return the dataframe
+    return distributed_aged_weight_proportions.filter( regex = '^(?!weight_|total_weight_|proportion_).*' )
 
-def calculate_aged_proportions( weight_length_age_sex_stratum: pd.DataFrame ,
-                                weight_strata: pd.DataFrame ):
+def calculate_aged_biomass( kriging_biomass_df: pd.DataFrame ,
+                            specimen_data: pd.DataFrame ,
+                            length_distribution: pd.DataFrame ,
+                            age_distribution: pd.DataFrame ,
+                            aged_proportions: pd.DataFrame ):
+    """
+    Calculate the kriged biomass distributed over length and age for each sex and among
+    all fish (sexed)
+
+    Parameters
+    ----------
+    kriging_biomass_df: pd.DataFrame
+        Dataframe containing kriged biomass estimates
+    specimen_data: pd.DataFrame
+        Dataframe containing specimen weights
+    length_distribution: np.ndarray
+        Array containing length bins/intervals
+    age_distribution: np.ndarray
+        Array containing age bins/intervals
+    aged_proportions: pd.DataFrame
+        Dataframe containing the weight proportions of aged fish within each stratum
+    """
+
+    ### Sum aged fish weights across age and length bins, then calculate weight proportions
+    proportions_weight_length_age_sex = compute_aged_weight_proportions( specimen_data ,
+                                                                         length_distribution ,
+                                                                         age_distribution )
     
-    ### Calculate aged / unaged proportions 
-    aged_proportions = ( 
-        weight_length_age_sex_stratum
-        .loc[ lambda df: df.sex != 'all' ] 
-        .groupby( [ 'stratum_num' ] ) 
-        .apply( lambda df: pd.Series( { 'weight_aged_total': df.summed_weight_all.sum( ) } ) )  
-        .reset_index( ) 
-        .merge( weight_strata , on = [ 'stratum_num' ] )  
-        .assign( proportion_aged_total = lambda x: x.weight_aged_total / x.weight_stratum_total , 
-                 proportion_unaged_total = lambda x: np.round( 1.0 - x.proportion_aged_total , decimals = 10 ) ) 
-    ) 
+    ### Calculate the summed aged proportions for all aged fish, and for aged fish 
+    # ---- belonging to each sex
+    aged_sex_proportions = compute_aged_sex_weight_proportions( proportions_weight_length_age_sex ,
+                                                                aged_proportions )
+    
+    ### Calculate weight proportions of aged fish distributed over age-length bins
+    # ---- for each sex within each stratum relative to the summed weights of each
+    # ---- stratum (i.e. aged + unaged weights)
+    distributed_sex_proportions = distribute_aged_weight_proportions( proportions_weight_length_age_sex ,
+                                                                      aged_sex_proportions )
```

Comment:
may be a good idea to make `distributed_sex_proportions` something longer, like `distributed_sex_length_age_proportions_wrt_all`